bdns/dns.go

```diff
@@ -427,17 +428,17 @@ func (dnsClient *impl) LookupHost(ctx context.Context, hostname string) ([]netip
 		// branching. We don't use ProblemDetails and SubProblemDetails here, because
 		// this error will get wrapped in a DNSError and further munged by higher
 		// layers in the stack.
-		return nil, resolvers, fmt.Errorf("%w; %s", errA, errAAAA)
+		return nil, resolvers, false, fmt.Errorf("%w; %s", errA, errAAAA)
 	}
 
-	return append(addrsA, addrsAAAA...), resolvers, nil
+	return append(addrsA, addrsAAAA...), resolvers, adA && adAAAA, nil
```

Comment:
This has somewhat confusing semantics in this case:

 - Response for `A` is successful, with AD bit set.
 - Response for `AAAA` is an error.

This results in returning false for the `ad` boolean.

This makes me think maybe LookupHost combining A and AAAA lookups is the wrong abstraction. In the VA, in `newHTTPValidationTarget` and `tryGetChallengeCert`, the first thing we do with the combined response is to split it into v4 addresses and v6 addresses. Perhaps we can simplify things with `LookupA` and `LookupAAAA` being separate.